### PR TITLE
ref(sampling): Remove error rules code - backend - [TET-103]

### PR DESCRIPTION
**Pull request by priscilawebdev**
continuation of PR https://github.com/getsentry/sentry/pull/35003

### CODE REVIEW

File: src/sentry/api/endpoints/project_details.py [hunk 1]
```diff
@@ -616,19 +610,6 @@ def put(self, request: Request, project) -> Response:
 
         if "dynamicSampling" in result:
             raw_dynamic_sampling = result["dynamicSampling"]
-            if (
```
Comment (andriisoldatenko) [inline src/sentry/api/endpoints/project_details.py:619]: do we want to remove tests?

```
/tests/sentry/api/endpoints/test_project_details.py
```
?
cc @ahmedetefy

Comment (priscilawebdev) [reply src/sentry/api/endpoints/project_details.py:619]: do you mean create new tests? because there are no tests checking this logic removed

Comment (ahmedetefy) [reply src/sentry/api/endpoints/project_details.py:619]: Usually, I would say yes that makes sense but looking at that test module, I don't think that branch was tested
@andriisoldatenko which tests specifically are you referring to ?

**Review by ahmedetefy — APPROVED** [review]
lgtm

**Pull request merged**